mesonbuild/interpreter/interpreterobjects.py

```diff
@@ -448,7 +448,6 @@ def name_method(self, args: T.List[TYPE_var], kwargs: TYPE_kwargs) -> str:
             ContainerTypeInfo(list, str, pairs=True),
             default=[],
             listify=True,
-            validator=lambda x: 'must be of length 2 or empty' if len(x) not in {0, 2} else None,
         ),
     )
     def pkgconfig_method(self, args: T.Tuple[str], kwargs: 'kwargs.DependencyPkgConfigVar') -> str:
```

Comment:
```suggestion
    def pkgconfig_method(self, args: T.Tuple[str], kwargs: 'kwargs.DependencyPkgConfigVar') -> str:
        if len(kwargs['define_variable'] > 2):
            FeatureNew.single_use('multiple pkg-config define overrides', '0.64', self.subproject, location=self.current_node)
```

And also in the `get_variable()` method